multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -528,9 +528,15 @@ def fit(
             )
 
         if self._output_shape is not None:
-            assert self._output_shape == output_shape, (
-                f"Inferred output shape {output_shape} is different from " f"the previous {self._output_shape}"
-            )
+            # Fix output shape inconsistency in countinuos training
+            if self._problem_type is not None and self._problem_type == NER and self._config is not None:
+                assert self._output_shape == (
+                    output_shape + len(OmegaConf.to_object(self._config.model.ner.special_tags))
```

Comment:
If `output_shape ` changes in continuous training, do we change the prediction head of the model accordingly?

Reply:
if output_shape changes it will raise an error

Reply:
It looks like here we allow the output_shape increases by `len(OmegaConf.to_object(self._config.model.ner.special_tags)`.

Reply:
If the output_shape actually doesn't change, that means the `infer_problem_type_output_shape` doesn't consider the special tags. I think we'd better to optimize `infer_problem_type_output_shape` rather than adding this confusing check.

Reply:
infer_problem_type_output_shape is used in fit() where the config is not obtained yet. The initialization of config is dependent on the infer_problem_type_output_shape function, if we add this line to infer_problem_type_output_shape it will have a dependent loop.

Reply:
resolved in [commit](https://github.com/awslabs/autogluon/pull/2232/commits/c71ce038499bfff90493cbb9afac904b2df5ca0a)